src/flb_log.c

```diff
@@ -454,6 +478,78 @@ int flb_log_set_file(struct flb_config *config, char *out)
     return 0;
 }
 
+
+/*
+ * Create and register cmetrics for the runtime logger.
+ * The caller must free the returned struct using flb_log_metrics_destroy.
+ */
+struct flb_log_metrics *flb_log_metrics_create()
+{
+    struct flb_log_metrics *metrics;
+    int i;
+    const char *message_type_str;
+    uint64_t ts;
+    int ret;
+
+    metrics = flb_calloc(1, sizeof(struct flb_log_metrics));
+    if (metrics == NULL) {
+        flb_errno();
+        return NULL;
+    }
+
+    metrics->cmt = cmt_create();
+    if (!metrics->cmt) {
+        flb_free(metrics);
+        return NULL;
+    }
+
+    metrics->logs_total_counter = cmt_counter_create(metrics->cmt,
+                                                     "fluentbit",
+                                                     "logger",
+                                                     "logs_total",
+                                                     "Total number of logs",
+                                                     1, (char *[]) {"message_type"});
+    if (metrics->logs_total_counter == NULL) {
+        cmt_destroy(metrics->cmt);
+        flb_free(metrics);
+        return NULL;
+    }
+
+    /*
+     * Initialize counters for all log message types to 0.
+     * This assumes types are contiguous starting at 1 (FLB_LOG_ERROR).
+     */
+    ts = cfl_time_now();
+    for (i = 1; ; i++) {
```

Comment:
Please change this loop so it explicitly iterates from `FLB_LOG_ERROR` up to `FLB_LOG_TRACE`.

Your mechanism is smart but not obvious at first glance and in order to minimize the chance of someone misunderstanding the code and making a mistake we need to make things painfully obvious when possible.

Reply:
Also, rename `i` to a proper name, we don't use single letter variables because it's a slippery slope that can lead to sub par and hard to follow code which can (and has) cause(d) issues.

Reply:
Updated.

I renamed the loop variable from `i` to `log_message_type`, though fwiw there is a lot of prior art in the code around using `i`.